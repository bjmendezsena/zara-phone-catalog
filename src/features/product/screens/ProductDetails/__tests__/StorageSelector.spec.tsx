import { render, getHtml, mockProductDetailsResponse } from "@/tests";
import {
  StorageSelector,
  StorageSelectorProps,
} from "@/features/product/screens/ProductDetails/StorageSelector";

describe(`<${StorageSelector.name}/>`, () => {
  const defaultProps: StorageSelectorProps = {
    storageOptions: mockProductDetailsResponse.storageOptions,
    selectedCapacity: mockProductDetailsResponse.storageOptions[0].capacity,
    onStorageChange: jest.fn(),
  };

  it("renders without crashing", () => {
    const { container } = render(<StorageSelector {...defaultProps} />);
    expect(getHtml(container)).toMatchSnapshot();
  });

  it("displays the correct number of storage options", () => {
    const { getAllByRole } = render(<StorageSelector {...defaultProps} />);
    const storageOptions = getAllByRole("button");
    expect(storageOptions).toHaveLength(defaultProps.storageOptions.length);
  });

  it("highlights the selected storage option", () => {
    const { getByTestId } = render(<StorageSelector {...defaultProps} />);
    const testId = `storage-option-${defaultProps.selectedCapacity}`;
    const selectedOption = getByTestId(testId);
    expect(selectedOption).toHaveClass("border-primary");
  });

  it("calls onStorageChange when a storage option is clicked", () => {
    const { getByTestId } = render(<StorageSelector {...defaultProps} />);
    const testId = `storage-option-${defaultProps.storageOptions[1].capacity}`;
    const storageOption = getByTestId(testId);
    storageOption.click();
    expect(defaultProps.onStorageChange).toHaveBeenCalledWith(
      defaultProps.storageOptions[1]
    );
  });
});
