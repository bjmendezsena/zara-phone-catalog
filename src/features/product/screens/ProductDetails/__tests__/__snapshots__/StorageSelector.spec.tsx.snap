// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<StorageSelector/> renders without crashing 1`] = `
"<div>
  <div>
    <h3
      class="text-sm  mb-4 uppercase tracking-wide"
    >
      STORAGE 
      <span
        class="uppercase"
      >
        How much space do you need?
      </span>
    </h3>
    <div
      class="flex divide-x-4"
    >
      <button
        aria-label="Select 128 GB storage"
        class="cursor-pointer px-6 py-3 border text-sm transition-colors hover:border-primary hover:border-r border-primary bg-primary text-white"
        data-testid="storage-option-128 GB"
      >
        128 GB
      </button>
      <button
        aria-label="Select 256 GB storage"
        class="cursor-pointer px-6 py-3 border text-sm transition-colors border-gray-300 text-secondary hover:border-primary hover:border-r border-r-gray-300"
        data-testid="storage-option-256 GB"
      >
        256 GB
      </button>
    </div>
  </div>
</div>"
`;
